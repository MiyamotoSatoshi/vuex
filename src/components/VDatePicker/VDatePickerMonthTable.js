// Mixins
import Colorable from '../../mixins/colorable'
import DatePickerTable from './mixins/date-picker-table'

// Utils
import { pad, createNativeLocaleFormatter } from './util'

export default {
  name: 'v-date-picker-month-table',

  mixins: [
    Colorable,
    DatePickerTable
  ],

  computed: {
    formatter () {
      return this.format || createNativeLocaleFormatter(this.locale, { month: 'short', timeZone: 'UTC' }, { start: 5, length: 2 })
    }
  },

  methods: {
    calculateTableDate (delta) {
      return `${parseInt(this.tableDate, 10) + Math.sign(delta || 1)}`
    },
    genTBody () {
      const children = []
      const cols = Array(3).fill(null)
      const rows = 12 / cols.length

      for (let row = 0; row < rows; row++) {
        const tds = cols.map((_, col) => {
          const month = row * cols.length + col
          return this.$createElement('td', {
            key: month
          }, [
            this.genButton(`${this.displayedYear}-${pad(month + 1)}`, false)
          ])
        })

        children.push(this.$createElement('tr', {
          key: row
        }, tds))
      }

      return this.$createElement('tbody', children)
    }
  },

  render (h) {
    return this.genTable('date-picker-table date-picker-table--month', [
      this.genTBody()
    ])
  }
}
