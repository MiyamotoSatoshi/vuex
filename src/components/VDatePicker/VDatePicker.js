// Components
import VDatePickerTitle from './VDatePickerTitle'
import VDatePickerHeader from './VDatePickerHeader'
import VDatePickerDateTable from './VDatePickerDateTable'
import VDatePickerMonthTable from './VDatePickerMonthTable'
import VDatePickerYears from './VDatePickerYears'

// Mixins
import Picker from '../../mixins/picker'

// Utils
import { pad, createNativeLocaleFormatter } from './util'
import isDateAllowed from './util/isDateAllowed'

/* @vue/component */
export default {
  name: 'v-date-picker',

  mixins: [Picker],

  props: {
    allowedDates: Function,
    // Function formatting the day in date picker table
    dayFormat: {
      type: Function,
      default: null
    },
    events: {
      type: [Array, Object, Function],
      default: () => null
    },
    eventColor: {
      type: [String, Function, Object],
      default: 'warning'
    },
    firstDayOfWeek: {
      type: [String, Number],
      default: 0
    },
    // Function formatting the tableDate in the day/month table header
    headerDateFormat: {
      type: Function,
      default: null
    },
    locale: {
      type: String,
      default: 'en-us'
    },
    max: String,
    min: String,
    // Function formatting month in the months table
    monthFormat: {
      type: Function,
      default: null
    },
    nextIcon: {
      type: String,
      default: '$vuetify.icons.next'
    },
    pickerDate: String,
    prevIcon: {
      type: String,
      default: '$vuetify.icons.prev'
    },
    reactive: Boolean,
    readonly: Boolean,
    scrollable: Boolean,
    showCurrent: {
      type: [Boolean, String],
      default: true
    },
    // Function formatting currently selected date in the picker title
    titleDateFormat: {
      type: Function,
      default: null
    },
    type: {
      type: String,
      default: 'date',
      validator: type => ['date', 'month'].includes(type) // TODO: year
    },
    value: String,
    // Function formatting the year in table header and pickup title
    yearFormat: {
      type: Function,
      default: null
    },
    yearIcon: String
  },

  data () {
    const now = new Date()
    return {
      activePicker: this.type.toUpperCase(),
      defaultColor: 'accent',
      inputDay: null,
      inputMonth: null,
      inputYear: null,
      isReversing: false,
      now,
      // tableDate is a string in 'YYYY' / 'YYYY-M' format (leading zero for month is not required)
      tableDate: (() => {
        if (this.pickerDate) {
          return this.pickerDate
        }

        const date = this.value || `${now.getFullYear()}-${now.getMonth() + 1}`
        const type = this.type === 'date' ? 'month' : 'year'
        return this.sanitizeDateString(date, type)
      })()
    }
  },

  computed: {
    current () {
      if (this.showCurrent === true) {
        return this.sanitizeDateString(`${this.now.getFullYear()}-${this.now.getMonth() + 1}-${this.now.getDate()}`, this.type)
      }

      return this.showCurrent || null
    },
    inputDate () {
      return this.type === 'date'
        ? `${this.inputYear}-${pad(this.inputMonth + 1)}-${pad(this.inputDay)}`
        : `${this.inputYear}-${pad(this.inputMonth + 1)}`
    },
    tableMonth () {
      return (this.pickerDate || this.tableDate).split('-')[1] - 1
    },
    tableYear () {
      return (this.pickerDate || this.tableDate).split('-')[0] * 1
    },
    minMonth () {
      return this.min ? this.sanitizeDateString(this.min, 'month') : null
    },
    maxMonth () {
      return this.max ? this.sanitizeDateString(this.max, 'month') : null
    },
    minYear () {
      return this.min ? this.sanitizeDateString(this.min, 'year') : null
    },
    maxYear () {
      return this.max ? this.sanitizeDateString(this.max, 'year') : null
    },
    formatters () {
      return {
        year: this.yearFormat || createNativeLocaleFormatter(this.locale, { year: 'numeric', timeZone: 'UTC' }, { length: 4 }),
        titleDate: this.titleDateFormat || this.defaultTitleDateFormatter
      }
    },
    defaultTitleDateFormatter () {
      const titleFormats = {
        year: { year: 'numeric', timeZone: 'UTC' },
        month: { month: 'long', timeZone: 'UTC' },
        date: { weekday: 'short', month: 'short', day: 'numeric', timeZone: 'UTC' }
      }

      const titleDateFormatter = createNativeLocaleFormatter(this.locale, titleFormats[this.type], {
        start: 0,
        length: { date: 10, month: 7, year: 4 }[this.type]
      })

      const landscapeFormatter = date => titleDateFormatter(date)
        .replace(/([^\d\s])([\d])/g, (match, nonDigit, digit) => `${nonDigit} ${digit}`)
        .replace(', ', ',<br>')

      return this.landscape ? landscapeFormatter : titleDateFormatter
    }
  },

  watch: {
    tableDate (val, prev) {
      // Make a ISO 8601 strings from val and prev for comparision, otherwise it will incorrectly
      // compare for example '2000-9' and '2000-10'
      const sanitizeType = this.type === 'month' ? 'year' : 'month'
      this.isReversing = this.sanitizeDateString(val, sanitizeType) < this.sanitizeDateString(prev, sanitizeType)
      this.$emit('update:pickerDate', val)
    },
    pickerDate (val) {
      if (val) {
        this.tableDate = val
      } else if (this.value && this.type === 'date') {
        this.tableDate = this.sanitizeDateString(this.value, 'month')
      } else if (this.value && this.type === 'month') {
        this.tableDate = this.sanitizeDateString(this.value, 'year')
      }
    },
    value () {
      this.setInputDate()
      if (this.value && !this.pickerDate) {
        this.tableDate = this.sanitizeDateString(this.inputDate, this.type === 'month' ? 'year' : 'month')
      }
    },
    type (type) {
      this.activePicker = type.toUpperCase()

      if (this.value) {
        const date = this.sanitizeDateString(this.value, type)
        this.$emit('input', this.isDateAllowed(date) ? date : null)
      }
    }
  },

  created () {
    if (this.pickerDate !== this.tableDate) {
      this.$emit('update:pickerDate', this.tableDate)
    }
    this.setInputDate()
  },

  methods: {
    isDateAllowed (value) {
      return isDateAllowed(value, this.min, this.max, this.allowedDates)
    },
    yearClick (value) {
      this.inputYear = value
      if (this.type === 'month') {
        this.tableDate = `${value}`
      } else {
        this.tableDate = `${value}-${pad(this.tableMonth + 1)}`
      }
      this.activePicker = 'MONTH'
      this.reactive && this.isDateAllowed(this.inputDate) && this.$emit('input', this.inputDate)
    },
    monthClick (value) {
      this.inputYear = parseInt(value.split('-')[0], 10)
      this.inputMonth = parseInt(value.split('-')[1], 10) - 1
      if (this.type === 'date') {
        this.tableDate = value
        this.activePicker = 'DATE'
        this.reactive && this.isDateAllowed(this.inputDate) && this.$emit('input', this.inputDate)
      } else {
        this.$emit('input', this.inputDate)
        this.$emit('change', this.inputDate)
      }
    },
    dateClick (value) {
      this.inputYear = parseInt(value.split('-')[0], 10)
      this.inputMonth = parseInt(value.split('-')[1], 10) - 1
      this.inputDay = parseInt(value.split('-')[2], 10)
      this.$emit('input', this.inputDate)
      this.$emit('change', this.inputDate)
    },
    genPickerTitle () {
      return this.$createElement(VDatePickerTitle, {
        props: {
          date: this.value ? this.formatters.titleDate(this.value) : '',
          selectingYear: this.activePicker === 'YEAR',
          year: this.formatters.year(`${this.inputYear}`),
          yearIcon: this.yearIcon,
          value: this.value
        },
        slot: 'title',
        style: this.readonly ? {
          'pointer-events': 'none'
        } : undefined,
        on: {
          'update:selectingYear': value => this.activePicker = value ? 'YEAR' : this.type.toUpperCase()
        }
      })
    },
    genTableHeader () {
      return this.$createElement(VDatePickerHeader, {
        props: {
          nextIcon: this.nextIcon,
          color: this.color,
          dark: this.dark,
          disabled: this.readonly,
          format: this.headerDateFormat,
          light: this.light,
          locale: this.locale,
          min: this.activePicker === 'DATE' ? this.minMonth : this.minYear,
          max: this.activePicker === 'DATE' ? this.maxMonth : this.maxYear,
          prevIcon: this.prevIcon,
          value: this.activePicker === 'DATE' ? `${this.tableYear}-${pad(this.tableMonth + 1)}` : `${this.tableYear}`
        },
        on: {
          toggle: () => this.activePicker = (this.activePicker === 'DATE' ? 'MONTH' : 'YEAR'),
          input: value => this.tableDate = value
        }
      })
    },
    genDateTable () {
      return this.$createElement(VDatePickerDateTable, {
        props: {
          allowedDates: this.allowedDates,
          color: this.color,
          current: this.current,
          dark: this.dark,
          disabled: this.readonly,
          events: this.events,
          eventColor: this.eventColor,
          firstDayOfWeek: this.firstDayOfWeek,
          format: this.dayFormat,
          light: this.light,
          locale: this.locale,
          min: this.min,
          max: this.max,
          tableDate: `${this.tableYear}-${pad(this.tableMonth + 1)}`,
          scrollable: this.scrollable,
          value: this.value
        },
        ref: 'table',
        on: {
          input: this.dateClick,
          tableDate: value => this.tableDate = value
        }
      })
    },
    genMonthTable () {
      return this.$createElement(VDatePickerMonthTable, {
        props: {
          allowedDates: this.type === 'month' ? this.allowedDates : null,
          color: this.color,
          current: this.current ? this.sanitizeDateString(this.current, 'month') : null,
          dark: this.dark,
          disabled: this.readonly,
          format: this.monthFormat,
          light: this.light,
          locale: this.locale,
          min: this.minMonth,
          max: this.maxMonth,
          scrollable: this.scrollable,
          value: (!this.value || this.type === 'month') ? this.value : this.value.substr(0, 7),
          tableDate: `${this.tableYear}`
        },
        ref: 'table',
        on: {
          input: this.monthClick,
          tableDate: value => this.tableDate = value
        }
      })
    },
    genYears () {
      return this.$createElement(VDatePickerYears, {
        props: {
          color: this.color,
          format: this.yearFormat,
          locale: this.locale,
          min: this.minYear,
          max: this.maxYear,
          value: `${this.tableYear}`
        },
        on: {
          input: this.yearClick
        }
      })
    },
    genPickerBody () {
      const children = this.activePicker === 'YEAR' ? [
        this.genYears()
      ] : [
        this.genTableHeader(),
        this.activePicker === 'DATE' ? this.genDateTable() : this.genMonthTable()
      ]

      return this.$createElement('div', {
        key: this.activePicker,
        style: this.readonly ? {
          'pointer-events': 'none'
        } : undefined
      }, children)
    },
    // Adds leading zero to month/day if necessary, returns 'YYYY' if type = 'year',
    // 'YYYY-MM' if 'month' and 'YYYY-MM-DD' if 'date'
    sanitizeDateString (dateString, type) {
      const [year, month = 1, date = 1] = dateString.split('-')
      return `${year}-${pad(month)}-${pad(date)}`.substr(0, { date: 10, month: 7, year: 4 }[type])
    },
    setInputDate () {
      if (this.value) {
        const array = this.value.split('-')
        this.inputYear = parseInt(array[0], 10)
        this.inputMonth = parseInt(array[1], 10) - 1
        if (this.type === 'date') {
          this.inputDay = parseInt(array[2], 10)
        }
      } else {
        this.inputYear = this.inputYear || this.now.getFullYear()
        this.inputMonth = this.inputMonth == null ? this.inputMonth : this.now.getMonth()
        this.inputDay = this.inputDay || this.now.getDate()
      }
    }
  },

  render () {
    return this.genPicker('v-picker--date')
  }
}
