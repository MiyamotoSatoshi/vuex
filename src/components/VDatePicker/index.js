import VDatePicker from './VDatePicker'

/* istanbul ignore next */
VDatePicker.install = function install (Vue) {
  Vue.component(VDatePicker.name, VDatePicker)
}

export default VDatePicker
