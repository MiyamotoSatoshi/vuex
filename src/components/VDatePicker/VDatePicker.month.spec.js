import Vue from 'vue'
import { test } from '~util/testing'
import VDatePicker from '~components/VDatePicker'

test('VDatePicker.js', ({ mount, compileToFunctions }) => {
  it('should emit input event on year click', async () => {
    const cb = jest.fn()
    const wrapper = mount(VDatePicker, {
      propsData: {
        value: '2013-05',
        type: 'month'
      },
      data: {
        activePicker: 'YEAR'
      }
    })

    wrapper.vm.$on('input', cb);
    wrapper.find('.picker--date__years li.active + li')[0].trigger('click')
    expect(cb).toBeCalledWith('2012-05')
  })

  it('should not emit input event on year click if month is not allowed', async () => {
    const cb = jest.fn()
    const wrapper = mount(VDatePicker, {
      propsData: {
        value: '2013-05',
        type: 'month',
        allowedDates: []
      },
      data: {
        activePicker: 'YEAR'
      }
    })

    wrapper.vm.$on('input', cb);
    wrapper.find('.picker--date__years li.active + li')[0].trigger('click')
    expect(cb).not.toBeCalled()
  })

  it('should emit input event on month click', async () => {
    const cb = jest.fn()
    const wrapper = mount(VDatePicker, {
      propsData: {
        value: '2013-05',
        type: 'month'
      }
    })

    wrapper.vm.$on('input', cb);
    wrapper.find('.picker--date__table tbody tr:first-child td:first-child button')[0].trigger('click')
    expect(cb).toBeCalledWith('2013-01')
  })

  it('should be scrollable', async () => {
    const wrapper = mount(VDatePicker, {
      propsData: {
        value: '2013-05',
        type: 'month',
        scrollable: true
      }
    })

    wrapper.find('.picker--date__table')[0].trigger('wheel')
    await wrapper.vm.$nextTick()
    expect(wrapper.vm.tableYear).toBe(2012)
  })

  it('should match snapshot with pick-month prop', () => {
    const wrapper = mount(VDatePicker, {
      propsData: {
        value: '2013-05-07',
        type: 'month'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match snapshot with allowed dates as array', () => {
    const wrapper = mount(VDatePicker, {
      propsData: {
        value: '2013-05',
        type: 'month',
        allowedDates: ['2013-01', '2013-03', '2013-05', '2013-07', 'invalid month']
      }
    })

    expect(wrapper.find('.picker--date__table tbody')[0].html()).toMatchSnapshot()
  })

  it('should match snapshot with allowed dates as function', () => {
    const wrapper = mount(VDatePicker, {
      propsData: {
        value: '2013-05',
        type: 'month',
        allowedDates: date => date.substr(6, 1) === '1'
      }
    })

    expect(wrapper.find('.picker--date__table tbody')[0].html()).toMatchSnapshot()
  })

  it('should match snapshot with allowed dates as object', () => {
    const wrapper = mount(VDatePicker, {
      propsData: {
        value: '2013-05',
        type: 'month',
        allowedDates: { min: '2013-03', max: '2013-07' }
      }
    })

    expect(wrapper.find('.picker--date__table tbody')[0].html()).toMatchSnapshot()
  })

  it('should match snapshot with month formatting functions', () => {
    const wrapper = mount(VDatePicker, {
      propsData: {
        value: '2005-11-01',
        type: 'month',
        monthFormat: date => `(${date.split('-')[1]})`
      }
    })

    expect(wrapper.find('.picker--date__table tbody')[0].html()).toMatchSnapshot()
  })

  it('should match snapshot with colored picker', () => {
    const wrapper = mount(VDatePicker, {
      propsData: {
        type: 'month',
        value: '2005-11-01',
        color: 'primary',
        headerColor: 'orange darken-1'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match snapshot with colored picker', () => {
    const wrapper = mount(VDatePicker, {
      propsData: {
        type: 'month',
        value: '2005-11-01',
        color: 'orange darken-1'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match change month when clicked on header arrow buttons', () => {
    const wrapper = mount(VDatePicker, {
      propsData: {
        value: '2005-11',
        type: 'month'
      }
    })

    const [leftButton, rightButton] = wrapper.find('.picker--date__header-selector button')

    leftButton.trigger('click')
    expect(wrapper.vm.tableDate).toBe('2004')

    rightButton.trigger('click')
    expect(wrapper.vm.tableDate).toBe('2006')
  })

  it('should match change active picker when clicked on month button', () => {
    const wrapper = mount(VDatePicker, {
      propsData: {
        value: '2005-11-01',
        type: 'month'
      }
    })

    const button = wrapper.find('.picker--date__header-selector-date strong')[0]

    button.trigger('click')
    expect(wrapper.vm.activePicker).toBe('YEAR')
  })

  it('should select year', async () => {
    const wrapper = mount(VDatePicker, {
      data: {
        activePicker: 'YEAR'
      },
      propsData: {
        type: 'month',
        value: '2005-11'
      }
    })

    wrapper.find('.picker--date__years li.active + li')[0].trigger('click')
    expect(wrapper.vm.activePicker).toBe('MONTH')
    expect(wrapper.vm.tableDate).toBe('2004')
  })

  it.skip('should calculate the first allowed month', () => {
    const today = new Date().toISOString().substr(0, 7)

    const wrapper1 = mount(VDatePicker, {
      propsData: {
        value: null,
        type: 'month'
      }
    })
    expect(wrapper1.vm.inputDate).toBe(today)

    // The behaviour is dependent on the current date
    // TODO refactor the test or change firstAllowedMonth implementation
    //
    // const allowedMonth = today.replace(/..$/, today.substr(5, 2) === '02' ? '03' : '02')
    // const wrapper2 = mount(VDatePicker, {
    //   propsData: {
    //     value: null,
    //     type: 'month',
    //     allowedDates: [allowedMonth]
    //   }
    // })
    // expect(wrapper2.vm.inputDate).toBe(allowedMonth)
  })

  it('should set the table date when value has changed', () => {
    const wrapper = mount(VDatePicker, {
      propsData: {
        value: null,
        type: 'month'
      }
    })

    wrapper.setProps({ value: '2005-11' })
    expect(wrapper.vm.tableDate).toBe('2005')
  })

  it('should update the active picker if type has changed', () => {
    const wrapper = mount(VDatePicker, {
      propsData: {
        type: 'month'
      }
    })

    expect(wrapper.vm.activePicker).toBe('MONTH')
    wrapper.setProps({ type: 'date' })
    expect(wrapper.vm.activePicker).toBe('MONTH')
    wrapper.setProps({ type: 'year' })
    expect(wrapper.vm.activePicker).toBe('YEAR')
  })

  it('should use prev and next icons', () => {
    const wrapper = mount(VDatePicker, {
      propsData: {
        type: 'month',
        prependIcon: 'block',
        appendIcon: 'check'
      }
    })

    const icons = wrapper.find('.picker--date__header-selector .icon')
    expect(icons[0].element.textContent).toBe('block')
    expect(icons[1].element.textContent).toBe('check')
  })
})
