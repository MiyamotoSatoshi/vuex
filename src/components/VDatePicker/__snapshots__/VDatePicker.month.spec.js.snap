// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VDatePicker.js should match snapshot with allowed dates as array 1`] = `

<tbody>
  <tr>
    <td>
      <button type="button"
              class="btn btn--flat"
      >
        <div class="btn__content">
          Jan
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="btn btn--flat btn--disabled"
              disabled
      >
        <div class="btn__content">
          Feb
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="btn btn--flat"
      >
        <div class="btn__content">
          Mar
        </div>
      </button>
    </td>
  </tr>
  <tr>
    <td>
      <button type="button"
              class="btn btn--flat btn--disabled"
              disabled
      >
        <div class="btn__content">
          Apr
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="btn btn--active btn--depressed accent"
      >
        <div class="btn__content">
          May
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="btn btn--flat btn--disabled"
              disabled
      >
        <div class="btn__content">
          Jun
        </div>
      </button>
    </td>
  </tr>
  <tr>
    <td>
      <button type="button"
              class="btn btn--flat"
      >
        <div class="btn__content">
          Jul
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="btn btn--flat btn--disabled"
              disabled
      >
        <div class="btn__content">
          Aug
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="btn btn--flat btn--disabled"
              disabled
      >
        <div class="btn__content">
          Sep
        </div>
      </button>
    </td>
  </tr>
  <tr>
    <td>
      <button type="button"
              class="btn btn--flat btn--disabled"
              disabled
      >
        <div class="btn__content">
          Oct
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="btn btn--flat btn--disabled"
              disabled
      >
        <div class="btn__content">
          Nov
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="btn btn--flat btn--disabled"
              disabled
      >
        <div class="btn__content">
          Dec
        </div>
      </button>
    </td>
  </tr>
</tbody>

`;

exports[`VDatePicker.js should match snapshot with colored picker 1`] = `

<div class="picker picker--date card"
     style="height: auto;"
     data-ripple="false"
>
  <div class="picker__title orange darken-1">
    <div class="date-picker-title">
      <div class="picker__title__btn date-picker-title__year">
        2005
      </div>
      <div class="picker__title__btn date-picker-title__date active">
        <div>
          November
        </div>
      </div>
    </div>
  </div>
  <div class="picker__body">
    <div>
      <div class="date-picker-header">
        <button type="button"
                class="btn btn--icon"
                data-ripple="true"
        >
          <div class="btn__content">
            <i aria-hidden="true"
               class="icon material-icons"
            >
              chevron_left
            </i>
          </div>
        </button>
        <div class="date-picker-header__value">
          <strong class="primary--text">
            2005
          </strong>
        </div>
        <button type="button"
                class="btn btn--icon"
                data-ripple="true"
        >
          <div class="btn__content">
            <i aria-hidden="true"
               class="icon material-icons"
            >
              chevron_right
            </i>
          </div>
        </button>
      </div>
      <div class="date-picker-table date-picker-table--month">
        <table>
          <tbody>
            <tr>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Jan
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Feb
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Mar
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Apr
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    May
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Jun
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Jul
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Aug
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Sep
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Oct
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Nov
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Dec
                  </div>
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>

`;

exports[`VDatePicker.js should match snapshot with colored picker 2`] = `

<div class="picker picker--date card"
     style="height: auto;"
     data-ripple="false"
>
  <div class="picker__title orange darken-1">
    <div class="date-picker-title">
      <div class="picker__title__btn date-picker-title__year">
        2005
      </div>
      <div class="picker__title__btn date-picker-title__date active">
        <div>
          November
        </div>
      </div>
    </div>
  </div>
  <div class="picker__body">
    <div>
      <div class="date-picker-header">
        <button type="button"
                class="btn btn--icon"
                data-ripple="true"
        >
          <div class="btn__content">
            <i aria-hidden="true"
               class="icon material-icons"
            >
              chevron_left
            </i>
          </div>
        </button>
        <div class="date-picker-header__value">
          <strong class="orange--text text--darken-1">
            2005
          </strong>
        </div>
        <button type="button"
                class="btn btn--icon"
                data-ripple="true"
        >
          <div class="btn__content">
            <i aria-hidden="true"
               class="icon material-icons"
            >
              chevron_right
            </i>
          </div>
        </button>
      </div>
      <div class="date-picker-table date-picker-table--month">
        <table>
          <tbody>
            <tr>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Jan
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Feb
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Mar
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Apr
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    May
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Jun
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Jul
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Aug
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Sep
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Oct
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Nov
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Dec
                  </div>
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>

`;

exports[`VDatePicker.js should match snapshot with month formatting functions 1`] = `

<tbody>
  <tr>
    <td>
      <button type="button"
              class="btn btn--flat"
      >
        <div class="btn__content">
          (01)
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="btn btn--flat"
      >
        <div class="btn__content">
          (02)
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="btn btn--flat"
      >
        <div class="btn__content">
          (03)
        </div>
      </button>
    </td>
  </tr>
  <tr>
    <td>
      <button type="button"
              class="btn btn--flat"
      >
        <div class="btn__content">
          (04)
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="btn btn--flat"
      >
        <div class="btn__content">
          (05)
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="btn btn--flat"
      >
        <div class="btn__content">
          (06)
        </div>
      </button>
    </td>
  </tr>
  <tr>
    <td>
      <button type="button"
              class="btn btn--flat"
      >
        <div class="btn__content">
          (07)
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="btn btn--flat"
      >
        <div class="btn__content">
          (08)
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="btn btn--flat"
      >
        <div class="btn__content">
          (09)
        </div>
      </button>
    </td>
  </tr>
  <tr>
    <td>
      <button type="button"
              class="btn btn--flat"
      >
        <div class="btn__content">
          (10)
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="btn btn--flat"
      >
        <div class="btn__content">
          (11)
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="btn btn--flat"
      >
        <div class="btn__content">
          (12)
        </div>
      </button>
    </td>
  </tr>
</tbody>

`;

exports[`VDatePicker.js should match snapshot with pick-month prop 1`] = `

<div class="picker picker--date card"
     style="height: auto;"
     data-ripple="false"
>
  <div class="picker__title primary">
    <div class="date-picker-title">
      <div class="picker__title__btn date-picker-title__year">
        2013
      </div>
      <div class="picker__title__btn date-picker-title__date active">
        <div>
          May
        </div>
      </div>
    </div>
  </div>
  <div class="picker__body">
    <div>
      <div class="date-picker-header">
        <button type="button"
                class="btn btn--icon"
                data-ripple="true"
        >
          <div class="btn__content">
            <i aria-hidden="true"
               class="icon material-icons"
            >
              chevron_left
            </i>
          </div>
        </button>
        <div class="date-picker-header__value">
          <strong class="accent--text">
            2013
          </strong>
        </div>
        <button type="button"
                class="btn btn--icon"
                data-ripple="true"
        >
          <div class="btn__content">
            <i aria-hidden="true"
               class="icon material-icons"
            >
              chevron_right
            </i>
          </div>
        </button>
      </div>
      <div class="date-picker-table date-picker-table--month">
        <table>
          <tbody>
            <tr>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Jan
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Feb
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Mar
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Apr
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    May
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Jun
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Jul
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Aug
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Sep
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Oct
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Nov
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="btn btn--flat"
                >
                  <div class="btn__content">
                    Dec
                  </div>
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>

`;
